diff --git a/gpt2_gpipe_memory_profile.py b/gpt2_gpipe_memory_profile.py
index c013d65..18a2b4a 100644
--- a/gpt2_gpipe_memory_profile.py
+++ b/gpt2_gpipe_memory_profile.py
@@ -26,6 +26,8 @@ from rich.cells import cell_len
 from rich.style import Style
 import pyfiglet
 import pickle
+import wandb
+
 
 console = Console()
 
@@ -260,6 +262,10 @@ def train():
 
 
 if __name__ == "__main__":
+    
+    with wandb.init(project="my-project", save_code=True) as run:
+        # your training code here
+        run.log({"loss": 0.123})
     console.print(Text(pyfiglet.figlet_format("start-main", font="slant"), style="bold cyan"))
     torch.cuda.memory._record_memory_history()
     train()
